ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -154,11 +159,19 @@ public KsqlResource(
     this.registerKsqlStatementTasks();
   }
 
+  // CHECKSTYLE_RULES.OFF: NPathComplexity
   @POST
   public Response handleKsqlStatements(final KsqlRequest request) {
+    // CHECKSTYLE_RULES.ON: NPathComplexity
     final List<PreparedStatement> parsedStatements;
     final KsqlEntityList result = new KsqlEntityList();
 
+    try {
+      waitForCommandOffset(request.getCommandOffset());
```

Comment:
We should factor this out to a common place and add similar logic to StreamedQueryResource and the websocket endpoint (WSQueryEndpoint).